<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>795</width>
    <height>505</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Rail Editor</string>
  </property>
  <widget class="QWidget" name="ui_centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QGroupBox" name="ui_sourceGroupBox">
      <property name="title">
       <string>Source</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="EditTableWidget" name="ui_sourceEditTableWidget">
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAsNeeded</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="textElideMode">
          <enum>Qt::ElideMiddle</enum>
         </property>
         <property name="showGrid">
          <bool>true</bool>
         </property>
         <property name="gridStyle">
          <enum>Qt::DotLine</enum>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QGroupBox" name="ui_writeDirectionGroupBox">
           <property name="title">
            <string>Write Direction</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QRadioButton" name="ui_horizontalDirectionRadioButton">
              <property name="text">
               <string>Horizontal</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="ui_verticalDirectionRadioButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Vertical</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="ui_smartDirectionRadioButton">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Smart</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QLabel" name="ui_rowValueLabel">
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
         <item row="0" column="6">
          <widget class="QLabel" name="ui_columnValueLabel">
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <spacer name="ui_sourceSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="3">
          <widget class="QLabel" name="ui_rowLabel">
           <property name="text">
            <string>Row</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
         <item row="0" column="5">
          <widget class="QLabel" name="ui_columnLabel">
           <property name="text">
            <string>Column</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QGroupBox" name="ui_insertModeGroupBox">
           <property name="title">
            <string>Insert Mode</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QRadioButton" name="ui_overRadioButton">
              <property name="text">
               <string>OVER</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="ui_insertRadioButton">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>INSERT</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="ui_smartInserRadioButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>SMART</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="ui_menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>795</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="ui_menuMenu">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="ui_newFileAction"/>
    <addaction name="ui_openFileAction"/>
    <addaction name="ui_saveFileAction"/>
    <addaction name="ui_saveAsAction"/>
    <addaction name="separator"/>
    <addaction name="ui_quitAction"/>
   </widget>
   <widget class="QMenu" name="ui_editMenu">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="ui_undoAction"/>
    <addaction name="ui_redoAction"/>
    <addaction name="separator"/>
    <addaction name="ui_cutAction"/>
    <addaction name="ui_copyAction"/>
    <addaction name="ui_pasteAction"/>
   </widget>
   <widget class="QMenu" name="ui_compileMenu">
    <property name="title">
     <string>Compile</string>
    </property>
    <addaction name="ui_setInterpreterAction"/>
    <addaction name="ui_runInterpreterAction"/>
    <addaction name="ui_stopInterpreterAction"/>
    <addaction name="separator"/>
    <addaction name="ui_setBuildAction"/>
    <addaction name="ui_runBuildAction"/>
    <addaction name="ui_stopBuildAction"/>
    <addaction name="separator"/>
    <addaction name="ui_buildAndRunJavaAction"/>
    <addaction name="ui_runJavaAction"/>
    <addaction name="ui_stopJavaAction"/>
   </widget>
   <widget class="QMenu" name="ui_windowMenu">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="ui_preferencesAction"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="ui_menuMenu"/>
   <addaction name="ui_editMenu"/>
   <addaction name="ui_compileMenu"/>
   <addaction name="ui_windowMenu"/>
  </widget>
  <widget class="QStatusBar" name="ui_statusBar"/>
  <widget class="QDockWidget" name="ui_ioDockWidget">
   <property name="windowTitle">
    <string>Interpreter I/O</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="3" column="0">
      <widget class="QPlainTextEdit" name="ui_outputPlainTextEdit">
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPlainTextEdit" name="ui_inputPlainTextEdit">
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="ui_inputLabel">
       <property name="text">
        <string>Input</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="ui_outpuLabel">
       <property name="text">
        <string>Output</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="ui_compilerDockWidget">
   <property name="windowTitle">
    <string>Compiler Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QPlainTextEdit" name="ui_compilerOutputPlainTextEdit">
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="centerOnScroll">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="ui_consoleDockWidget">
   <property name="windowTitle">
    <string>Application Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="ApplicationConsole" name="ui_consolePlainTextEdit">
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="centerOnScroll">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="ui_issuesDockWidget">
   <property name="windowTitle">
    <string>Issues</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_6">
     <item row="0" column="0">
      <widget class="QListWidget" name="ui_issuesListWidget">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="ui_menuToolBar">
   <property name="windowTitle">
    <string>Menu Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="ui_newFileAction"/>
   <addaction name="ui_openFileAction"/>
   <addaction name="ui_saveFileAction"/>
  </widget>
  <widget class="QToolBar" name="ui_compileToolBar">
   <property name="windowTitle">
    <string>Build Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="ui_runBuildAction"/>
   <addaction name="ui_buildAndRunJavaAction"/>
   <addaction name="ui_stopJavaAction"/>
  </widget>
  <widget class="QToolBar" name="ui_editToolBar">
   <property name="windowTitle">
    <string>Edit Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="ui_undoAction"/>
   <addaction name="ui_redoAction"/>
   <addaction name="separator"/>
   <addaction name="ui_cutAction"/>
   <addaction name="ui_copyAction"/>
   <addaction name="ui_pasteAction"/>
  </widget>
  <action name="ui_newFileAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/New File.png</normaloff>:/icons/Buttons/New File.png</iconset>
   </property>
   <property name="text">
    <string>New File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_openFileAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Open File.png</normaloff>:/icons/Buttons/Open File.png</iconset>
   </property>
   <property name="text">
    <string>Open File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_saveFileAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Save File.png</normaloff>:/icons/Buttons/Save File.png</iconset>
   </property>
   <property name="text">
    <string>Save File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_quitAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Quit.png</normaloff>:/icons/Buttons/Quit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q, Alt+X</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_saveAsAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Save As.png</normaloff>:/icons/Buttons/Save As.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_setInterpreterAction">
   <property name="text">
    <string>Set Interpreter Location</string>
   </property>
  </action>
  <action name="ui_runInterpreterAction">
   <property name="text">
    <string>Run Interpreter</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_undoAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Undo.png</normaloff>:/icons/Buttons/Undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_redoAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Redo.png</normaloff>:/icons/Buttons/Redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_stopInterpreterAction">
   <property name="text">
    <string>Stop Interpreter</string>
   </property>
  </action>
  <action name="ui_setBuildAction">
   <property name="text">
    <string>Set Compiler Location</string>
   </property>
  </action>
  <action name="ui_runBuildAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Build.png</normaloff>:/icons/Buttons/Build.png</iconset>
   </property>
   <property name="text">
    <string>Build</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Build&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="ui_stopBuildAction">
   <property name="text">
    <string>Cancel Build</string>
   </property>
  </action>
  <action name="ui_cutAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Cut.png</normaloff>:/icons/Buttons/Cut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="ui_copyAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Copy.png</normaloff>:/icons/Buttons/Copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="ui_pasteAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Paste.png</normaloff>:/icons/Buttons/Paste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="ui_buildAndRunJavaAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Run.png</normaloff>:/icons/Buttons/Run.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="ui_runJavaAction">
   <property name="text">
    <string>Run Without Build</string>
   </property>
  </action>
  <action name="ui_stopJavaAction">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/icons/Buttons/Stop.png</normaloff>:/icons/Buttons/Stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="ui_preferencesAction">
   <property name="text">
    <string>Preferences</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EditTableWidget</class>
   <extends>QTableWidget</extends>
   <header location="global">EditTableWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ApplicationConsole</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">ApplicationConsole.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
